virtwho/runner.py

```diff
@@ -68,207 +69,111 @@ def run_virtwho_service(self, wait=None):
         :return: tty_output and rhsm_output
         """
-        tty_output, rhsm_output = self.vw_start(wait=wait)
-        return tty_output, rhsm_output
+        rhsm_data = self.virtwho_start(wait=wait)
+        return rhsm_data
 
-    def log_analyzer(self,
-                     rhsm_log,
-                     send_number=1,
-                     error_number=0,
-                     error_list=None,
-                     report_id=None,
-                     interval_time=None,
-                     loop_number=None,
-                     loop_time=None):
-        """
-
-        :param rhsm_log: the log to be analyzed.
-        :param send_number:
-        :param error_number:
-        :param error_list:
-        :param report_id:
-        :param interval_time:
-        :param loop_number:
-        :param loop_time:
-        :return:
-        """
-        analyzer_list = list()
-        if send_number:
-            send_number_get = self.vw_send_number(rhsm_log)
-            logger.info(f'The send number get from log is {send_number_get}')
-            analyzer_list.append(send_number_get == send_number)
-        if error_number:
-            error_number_get = self.vw_error_number(rhsm_log)
-            logger.info(f'The error number get from log is {error_number_get}')
-            if error_number == 'nonzero' or error_number == 'nz':
-                analyzer_list.append(error_number_get > 0)
-            analyzer_list.append(error_number_get == error_number)
-        if error_list:
-            error_list_get = self.vw_error_list()
-            logger.info(f'The error list get from log is {error_list_get}')
-            analyzer_list.append(
-                self.msg_search(str(error_list_get), error_list)
-            )
-        if report_id:
-            report_id_get = self.vw_report_id(rhsm_log)
-            logger.info(f'The report id get from log is {report_id_get}')
-            analyzer_list.append(report_id_get == report_id)
-        if interval_time:
-            interval_time_get = self.vw_interval_time(rhsm_log)
-            logger.info(f'The interval time get from log is '
-                        f'{interval_time_get}')
-            analyzer_list.append(interval_time_get == interval_time)
-        if loop_number:
-            loop_number_get = self.vw_loop_number()[1]
-            logger.info(f'The loop number get from log is {loop_number_get}')
-            analyzer_list.append(loop_number_get == loop_number)
-        if loop_time:
-            loop_time_get = self.vw_loop_time()
-            logger.info(f'The loop time get from log is {loop_time_get}')
-            analyzer_list.append(loop_time_get == loop_time)
-        if 'False' in analyzer_list:
-            return False
-        else:
-            return True
+    def log_analyzer(self, rhsm_log):
+        data = dict()
+        data["debug"] = self.msg_search(rhsm_log, "\\[.*DEBUG\\]")
+        data["oneshot"] = self.msg_search(rhsm_log, "virt-who terminated")
```

Comment:
for the oneshot validation, I think only 'virt-who terminated' is not enough, sometimes, virt-who will be terminated by errors.

Reply:
Yes, it's really true, now my thought is to add asserting ```result["send_number"] == 1 and result["error_number"] == 0``` when test oneshot function.
And besides above I think only check the key words ```virt-who terminated``` is also not enough, so added ```data["thread_number"] = self.virtwho_thread_number()``` to check ```result["thread_number"] == 0``` to make sure virt-who service is really stopped.